service/history/commandChecker.go

```diff
@@ -509,6 +514,11 @@ func (v *commandAttrValidator) validateContinueAsNewWorkflowExecutionAttributes(
 	if err != nil {
 		return err
 	}
+
+	searchattribute.SetType(
+		attributes.GetSearchAttributes(),
+		searchattribute.ConvertDynamicConfigToIndexedValueTypes(v.config.ValidSearchAttributes()))
+
```

Comment:
why not do this logic within mutable state?

Reply:
Agree!